### ISSUE TITLE
fileName in Response is null, not undefined, contradicting Flow type

### ISSUE THREAD

**Issue by chrisbobbe** [react-native-image-picker/react-native-image-picker#1271]
I'm using 0.26.10, but the [type for `ImagePickerResponse.fileName` in the current version](https://github.com/react-native-community/react-native-image-picker/blob/master/src/internal/types.ts) has not changed:

```
export interface ImagePickerResponse {
  // [other things...]
  fileName?: string;
}
```

At least in the old version I'm using, fileName is sometimes `null`, which goes against `filename?: string;` because it is neither absent, undefined, nor a string.

If I upgrade to the most recent version, will this be fixed by fileName being undefined instead of null? (If so, feel free to close this right away.) Or is this type still incorrect?

The snippet above is TypeScript, but I believe that `index.js.flow` in the npm package is generated from that file.

**Comment by chrisbobbe** [comment]
Maybe the documentation at https://github.com/react-native-community/react-native-image-picker/blob/master/docs/Reference.md#the-response-object might be made a bit clearer, with exactly what happens if a value is "not available" -- i.e., is it null, undefined, empty string, etc., if someone has the time to look through the code :)

**Comment by Johan-dutoit** [comment]
Is this solved by https://github.com/react-native-community/react-native-image-picker/pull/1274?

**Comment by gnprice** [comment]
> Is this solved by #1274?

That appears to be about the `type` property, not `fileName`.

**Comment by eng-osman** [comment]
I have that problem too but ios fileName: null

**Comment by ldh1428a** [comment]
i have same problem anybody cannot solved this problem ?

**Comment by sujayjaju** [comment]
We handle it like this:
```
response.fileName || 'upload.png',
```

**Comment by tastafur** [comment]
i have that problem too in ios

**Comment by sujayjaju** [comment]
Workaround:
```
const ext = response.type ? response.type.split('/').pop() : 'png';
....
response.fileName || `upload.${ext}`,
```

**Comment by GunnarAK** [comment]
One other way that seems to work for me
```
let path = response.uri;
if (Platform.OS === "ios") {
   path = "~" + path.substring(path.indexOf("/Documents"));
}
if (!response.fileName) response.fileName = path.split("/").pop();
```

**Comment by vfa-giangnt** [comment]
worked for me! Tks in advanced

**Comment by Supun00SL** [comment]
finally working solution. thanks. @GunnarAK

**Comment by gnprice** [comment]
Stale-bot please go away - there's no sign this issue has been fixed.

**Comment by gnprice** [comment]
Stale-bot please go away -- there's no sign here this issue has been fixed.

**Comment by Pavneet-Sing** [comment]
Facing this issue on iOS 11.6 simulator, working fine for android.

**Comment by ravirajn22** [comment]
Please migrate to using version 3.x.x https://github.com/react-native-image-picker/react-native-image-picker/releases

Check 3.x.x docs for installation and API changes.
Post if you have any issue

**Comment by a-c-sreedhar-reddy** [comment]
@ravirajn22 is this version released? I see 2.3.4 as the latest version on npm

**Comment by ravirajn22** [comment]
https://github.com/react-native-image-picker/react-native-image-picker/releases
